tests/unittests/Foundation/NSTimerTests.mm

```diff
@@ -105,16 +105,15 @@ - (void)dealloc {
     NSTimerTestObj* testObj = [[[NSTimerTestObj alloc] initWithValue:NO] autorelease];
     volatile long waitCompletion = 0;
 
-    auto selectorCalledAsync = std::async(std::launch::async,
-                                          [&waitCompletion, testObj]() {
-                                              BOOL fooChanged = [testObj waitOnCalledConditionForInterval:15] && [testObj called];
-                                              _InterlockedExchange(&waitCompletion, 1L);
-                                              return fooChanged;
-                                          });
+    auto selectorCalledAsync = std::async(std::launch::async, [&waitCompletion, testObj]() {
+        BOOL fooChanged = [testObj waitOnCalledConditionForInterval:2] && [testObj called];
+        _InterlockedExchange(&waitCompletion, 1L);
+        return fooChanged;
+    });
 
     NSRunLoop* runLoop = [NSRunLoop currentRunLoop];
 
-    NSTimer* timer = [NSTimer scheduledTimerWithTimeInterval:10 target:testObj selector:@selector(testFunction) userInfo:nil repeats:NO];
+    NSTimer* timer = [NSTimer scheduledTimerWithTimeInterval:1 target:testObj selector:@selector(testFunction) userInfo:nil repeats:NO];
```

Comment:
Might be worth it to put some of these time intervals and other numbers at the top as statics to make updating this test easier in the future as well.